### DISCUSSION TITLE
Is it possible to do this with vico?

### DISCUSSION THREAD

**Discussion by ravenfeld** [patrykandpatrick/vico#536] [Questions] [answered]
Hello, I'm looking to do this as a graph, is it possible? 

Thank you in advance for your reply.

![Capture d’écran 2024-01-12 à 09 06 46](https://github.com/patrykandpatrick/vico/assets/3765114/c3dba93e-3d9d-4e4a-ab73-2d3d40895929)

**Comment by Gowsky** [comment, marked as answer]
Hi @ravenfeld, unfortunately, this is not possible. The API would require significant changes to support a chart like that.

**Reply by ravenfeld** [reply]
Thanks for your feedback, I'm going to draw directly via compose now.

**Reply by patrickmichalik** [reply]
Hello! To expand on this, while there’s no built-in API for such line backgrounds, you could implement a chart like this with Vico 2 by creating a custom `CartesianLayer`, possibly based on `LineCartesianLayer`. (Vico 1 could be used too, but it’s a bit less flexible in this respect.) Whether this would be quicker depends on the specifics of the desired result, however. Unless some of Vico’s more advanced features are required, using the Compose canvas is most likely quicker.

**Reply by ravenfeld** [reply]
No, I coded it myself using draw from jetpackcompose.

**Reply by patrickmichalik** [reply]
Yup, that’s what I was referring to in my comment’s last sentence. I just wanted to mention for completeness that it would be possible to implement this as a custom `CartesianLayer` (or, in Vico 1, a custom `Chart`).
